hydra-node/src/Hydra/Chain/Direct/Wallet.hs

```diff
@@ -183,11 +194,15 @@ withTinyWallet tracer magic (vk, sk) iocp addr action = do
           address
       , sign = \validatedTx@ValidatedTx{body, wits} ->
           let txid = Ledger.TxId (SafeHash.hashAnnotated body)
-              wit = txid `signWith` Ledger.KeyPair (Ledger.VKey vk) sk
+              wit =
+                fromLedgerTxId txid
+                  `signWith` ( Cardano.Api.PaymentVerificationKey (Ledger.VKey vk)
+                             , Cardano.Api.PaymentSigningKey sk
+                             )
```

Comment:
Should: Not require a keypair, but only a signing key to `signWith`.
The current signature allows to provide different vk than sk and is more a footgun than just taking the sk and deriving the vk from it.

Reply:
Indeed.